### PR TITLE
Add infrastructure for releasing deployment considerations as a fixed doc

### PR DESCRIPTION
**Pull request by tkfu**
Sample PDF output attached. Detailed instructions are in the readme, but as long as you have GNU make and docker, you should be able to just run `make pdf` or `make html`. That will give you an auto-generated version tag in the title. If you want to set a specific version tag, you can do `RELEASE_VERSION=foo make pdf` or `RELEASE_VERSION=foo make html`.

[uptane-deployment-considerations-release.pdf](https://github.com/uptane/deployment-considerations/files/5883210/uptane-deployment-considerations-release.pdf)

Pinging @iramcdonald because I know you wanted us to have a nice way to version deployment considerations docs.

### CODE REVIEW

**Comment by jhdalek55** [comment]
@mnm678 I'm going to let you review this first, since you have been working on this. Also, other than the README, it appears to be mostly code. @tkfu I assume this would automate the task?

I reviewed the PDF output very quickly and I do have things I would like modified. We need to add a Table of Contents (we don't have a markdown file for that but one could be added) And, we need to distinguish between Major Sections. I'd be happy to point out these things.

**Comment by pattivacek** [comment]
LGTM, with a few caveats:
* I agree with Lois about needing a table of contents.
* `Figure 3.How Primaries would interact with the Director repository` is weirdly off-centered.
* The changelog at the end doesn't have a header, and in the previously generated version, that was at the start of the document. I'm not invested in a specific ordering, but if there were specific choices made for good reasons, we should be consistent.
* The text about being a Linux Foundation JDF project is missing. I don't think we ever decided on a perfect place for that, but @jhdalek55 did something to pull that in before.

**Comment by tkfu** [comment]
Yup, we can iterate and improve on the final output format. This PR isn't about making sure we can generate a ready-to-release document, it's just to get the necessary tooling into the repo so that we can work towards getting the output we want.

**Comment by jhdalek55** [comment]
@tkfu I understand that this PR is about the mechanism and not the content. Just interjecting here that there was an agreed upon Table of Contents for this document that must be included. No need to do anything about it just now, but when you are ready to actually release this document, some of the things I mentioned and that @pattivacek pointed out will need tp be addressed.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+help: ## Print this message and exit
+	@echo "\033[1;37mRequires Docker\033[0m"
+	@awk 'BEGIN {FS = ":.*?## "} /^[0-9a-zA-Z_-]+:.*?## / {printf "\033[36m%s\033[0m : %s\n", $$1, $$2}' $(MAKEFILE_LIST) \
+		| column -s ':' -t
+
+html: ## Create an HTML version of the deployment considerations, using docker
+	@docker run --rm -it -v $(PWD):/data uptane/pandoc $(MKD) --filter pandoc-include --metadata=title:"Uptane Deployment Considerations v.$(RELEASE_VERSION)" -o $(HTML) --self-contained
+
+pdf: ## Create a PDF version of the deployment considerations, using docker
```
Comment (mnm678) [inline Makefile:27]: I think the `--toc` option will add a generated table of contents

**Review by mnm678 — APPROVED** [review]
Thanks for putting this together @tkfu! This should make releases much easier

**Comment by jhdalek55** [comment]
At the risk of sounding a bit foolish, @tkfu what specifically did you want me to review (since this PR is largely formatting code)?

**Comment by tkfu** [comment]
Oh, I didn't notice you were a requested reviewer @jhdalek55. I think github added you by default and I didn't see it. Feel free to ignore.

@mnm678, thanks for the tip--I added that to the makefile.

**Pull request merged**